posthog/integrations/django.py

```diff
@@ -0,0 +1,116 @@
+from typing import TYPE_CHECKING, cast
+from posthog import scopes
+
+if TYPE_CHECKING:
+    from django.http import HttpRequest, HttpResponse  # noqa: F401
+    from typing import Callable, Dict, Any, Optional  # noqa: F401
+
+
+class PosthogContextMiddleware:
+    """Middleware to automatically track Django requests.
+
+    This middleware wraps all calls with a posthog scope. It attempts to extract the following from the request headers:
+    - Session ID as $session_id, (extracted from `X-POSTHOG-SESSION-ID`)
+    - Distinct ID as $distinct_id, (extracted from `X-POSTHOG-DISTINCT-ID`)
+    - Request URL as $current_url
+    - Request Method as request_method
+
+    The scope will also auto-capture exceptions and send them to PostHog, unless you disable it by setting
+    `POSTHOG_MW_CAPTURE_EXCEPTIONS` to `False` in your Django settings.
+
+    The middleware behaviour is customisable through 3 additional functions:
+    - `POSTHOG_MW_EXTRA_TAGS`, which is a Callable[[HttpRequest], Dict[str, Any]] expected to return a dictionary of additional tags to be added to the scope.
```

Comment:
Should we be calling them `properties`? Tags are not a concept we use / mention elsewhere
```suggestion
    - `POSTHOG_MW_EXTRA_PROPERTIES`, which is a Callable[[HttpRequest], Dict[str, Any]] expected to return a dictionary of additional tags to be added to the scope.
```

Reply:
I used tags because they're scope/context tags (which we do call `tags()`)... I'll have a think about naming a bit more (I also mix "scope" and "context" in a way that's unhelpful), and either change it here or, more likely, make some breaking change in the next major version bump (which will come in a few days).

Reply:
Updated the comment to talk in terms of tags and contexts, included a note about how context tags end up as properties on events.